library/rtsp/src/main/java/com/google/android/exoplayer2/source/rtsp/RtspClient.java

```diff
@@ -288,6 +290,10 @@ private void maybeLogMessage(List<String> message) {
   /** Returns a {@link Socket} that is connected to the {@code uri}. */
   private static Socket getSocket(Uri uri) throws IOException {
     checkArgument(uri.getHost() != null);
+    if (uri.getScheme().equals("rtsps")) {
+      int rtspsPort = uri.getPort() > 0 ? uri.getPort() : DEFAULT_RTSPS_PORT;
+      return SSLSocketFactory.getDefault().createSocket(checkNotNull(uri.getHost()), rtspsPort);
+    }
```

Comment:
Would it make sense to generalize it a bit to:

```
boolean isRtsps = uri.getScheme().equals("rtsps");
int port = uri.getPort() > 0 
   ? uri.getPort()
   : isRtsps
     ? DEFAULT_RTSPS_PORT
     : DEFAULT_RTSP_PORT;

SocketFactory factory = isRtsps 
  ? SSLSocketFactory.getDefault()
  : SocketFactory.getDefault();

return factory.createSocket(checkNotNull(uri.getHost()), port)
```

Reply:
Thanks for the feedback, I updated the function now accordingly!